Loop/Managers/LoopDataManager.swift

```diff
@@ -1634,6 +1634,34 @@ extension LoopDataManager {
             handler(self, LoopStateView(loopDataManager: self, updateError: updateError))
         }
     }
+    
+    func generateSimpleBolusRecommendation(carbs: HKQuantity?, glucose: HKQuantity?) -> HKQuantity? {
+        var activeInsulin: Double? = nil
+        let semaphore = DispatchSemaphore(value: 0)
+        doseStore.insulinOnBoard(at: Date()) { (result) in
+            if case .success(let iobValue) = result {
+                activeInsulin = iobValue.value
+            }
+            semaphore.signal()
+        }
+        semaphore.wait()
```

Comment:
Is it possible for `insulinOnBoard` to never come back?  Or take too long? Should a timeout be considered here when calling `wait`?

Reply:
Short of an unexpected lockup in the code, no. Both success and error paths should return here in a reasonable amount of time.

Reply:
Ok...I now notice this is done a lot in the code. Not now, but I think eventually it would be a good idea to add something like this someday:
```swift
extension DispatchSemaphore {
    static let safeTimeout = DispatchTimeInterval.seconds(10)
    func safeWait(file: StaticString = #file, line: UInt = #line) {
        let result = self.wait(timeout: .now() + DispatchSemaphore.safeTimeout)
        precondition(result == .success, "Timed out waiting for semaphore", file: file, line: line)
    }
}
```
and use `safeWait()` in all places instead of `wait()`.  Then we'll get crash reports with stack traces when this occurs.  Just a suggestion :).